airflow/providers/salesforce/hooks/salesforce.py

```diff
@@ -236,6 +236,61 @@ def write_object_to_file(self,
         if fmt not in ['csv', 'json', 'ndjson']:
             raise ValueError("Format value is not recognized: {}".format(fmt))
 
+        if filename is None:
+            raise ValueError("Filename cannot be None.")
+
+        df = self.object_to_df(query_results=query_results, coerce_to_timestamp=coerce_to_timestamp,
+                               record_time_added=record_time_added)
```

Comment:
Why does this function needs to be called here? Can't the function not just be called outside of it?
I mean in the operator before `write_object_to_file` gets called.